### PR TITLE
Make dependency on unix relaxed

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

**Comment by phadej** [comment]
Build fails due GHC 7.2.1, I opened transformers issue: http://hub.darcs.net/ross/transformers/issue/16

**Comment by phadej** [comment]
Ok, it can be fixed, but to *trust* base, one would need sudo on travis.

**Comment by RyanGlScott** [comment]
I'll be away from my computer until Sunday, but I can look into this further then.

**Comment by RyanGlScott** [comment]
It does appear that that fix would prevent us from using containerized Travis builds due to `sudo`, but I don't think that would be a dealbreaker. (They've hinted that containerized builds might be able to use `sudo` [one day](https://docs.travis-ci.com/user/migrating-from-legacy/#Using-sudo-isn%E2%80%99t-possible-%28right-now%29), so it might just be a temporary workaround.)

I'm curious, though: is there a particular reason you need this need? I was under the impression that `unix` was always installed on all non-Windows OSes, effectively making it a non-dependency.

File: base-compat.cabal [hunk 1]
```diff
@@ -44,12 +44,21 @@ source-repository head
...
   ghc-options:
       -Wall
   build-depends:
       base >= 4.3 && < 5
   if !os(windows)
+      if flag(needs-unix)
+          build-depends: base >= 4.3 && <4.7, unix
+      else
+          build-depends: base >= 4.7 && <5
       build-depends: unix
```
Comment (RyanGlScott) [inline base-compat.cabal]: Would this line be redundant due to the above changes?

**Comment by phadej** [comment]
@RyanGlScott the change is motivated by fact, that it would be nice to test new `time-1.6`, but it cannot be really upgraded without force-reinstall of `unix` atm (as `unix` depends on `time`). Probably there will be `unix` release with `time-1.6` support soon, but aren't at the moment.

**Comment by RyanGlScott** [comment]
I wonder if `base-compat` is really the issue here. As you said, the upper bound on `time` from the `unix` package is the real cause. Can you just install the HEAD version of `unix` [here](https://github.com/haskell/unix) until a new version is uploaded to Hackage?

I'm hesitant to introduce conditional code paths into `base-compat.cabal`, since they require downstream users to make sense of our build flags and tend to be a maintenance headache down the road—especially if we need to change `System.Environment.Compat`.

**Comment by phadej** [comment]
@RyanGlScott **automatic** flags are they  way to say `(base >= 4.3 && <4.7, unix) || (base >= 4.7 && <5)`, no-one should set them manually (except in e.g. stack.yaml situations).

In some sense this is `base-compat` issue, as above disjunction is more precise dependency declaration.

I don't really want to complicate e.g. travis builds with `HEAD` `unix`. I really like `base-compat`, but as dependencies are unprecise, it unnecessarily restricts (though undirectly) me from already supporting `time-1.6`. I'm also unaware of `unix` release plan, so I don't know how long the situation will persist.

**Comment by phadej** [comment]
@hvr was kind, and relaxed `unix`'s (and `directory`'s) `time` bounds, so this PR is not important for me anymore. @RyanGlScott, use your judgement, you can just ignore this PR.

**Pull request closed without merging**